### ISSUE TITLE
It does not work with multiple parallel queries

### ISSUE THREAD

**Issue by mitar** [Richie765/pg-query-observer#1]
Initialization code has some race conditions and creating/dropping functions and triggers interfere if you start multiple triggers on the same query but different parameters at the same time, because code tries to do changes multiple times on same source tables.

Moreover, even when that is fixed, only one query really gets updates about changes, others do not.

I made ugly workarounds: https://github.com/mitar/pg-table-observer

**Comment by Richie765** [comment]
Thanks for your bug-report and also great work on the thorough benchmark test! I'm currently not using this package intensively so I cannot justify working on it at the moment. But I hope this can serve as a basis for others for their own work.

B.t.w. (just for information) for a new project I used a different method to handle reactivity. I moved all database access to an API. The API also has methods for subscribing to changes. Then when an API that modifies the DB is called, all the subscribers are notified as well.

I found this method easier to manage and it also allows to use any database server or any other storage method since it doesn't require tricks like triggers/notifications. But of course it depends on your project what is the best approach.

**Comment by mitar** [comment]
Yes, using APIs is cool. I used that in Django, where they have signals on any data modification. So you can also use it to implement reactivity.
